src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2500,6 +2500,9 @@ std::string Creature::GetScriptName() const
 
 uint32 Creature::GetScriptId() const
 {
+    if (CreatureData const* creatureData = GetCreatureData())
```

Comment:
that is pretty much hiding the template scripts, afaik, since CreatureData is never null for db spawns. you should check `scriptID != 0` rather than a null pointer. same for gameobjects